src/source_drivers/modbus/services/modbus_functions/polling/poll.py

```diff
@@ -156,10 +156,30 @@ def poll_point(network, device, point, transport) -> None:
             print(f'MODBUS ERROR: in poll main function {str(e)}')
         fault = True
         fault_message = str(e)
-    if not point_store:
-        point_store = PointStoreModel(fault=fault, fault_message=fault_message,
-                                      point_uuid=point.uuid)
+    if not point_store_new:
+        point_store_new = PointStoreModel(value=0, fault=fault, fault_message=fault_message, point_uuid=point.uuid)
     if modbus_debug_poll:
-        print("!!! END MODBUS POLL @@")
+        print("!!! END MODBUS POLL @@@")
 
-    return point_store
+    update_point_store(network, device, point, point_store_new)
+
+
+def update_point_store(network, device, point, point_store_new):
+    """
+    It compares :param {PointStoreModel} point_store_new with the existing point_store value for that particular point &
+    If new: update it and also call add_point_history_on_cov
+    If old: do nothing
+    """
+    if point_store_new.update():
+        add_point_history_on_cov(network, device, point)
+
+
+def add_point_history_on_cov(network, device, point):
+    """
+    add point.point_store to the point_store_history if they history type is 'COV' and history_enable is `True`
+    """
+    if point.history_type == HistoryType.COV \
+            and network.history_enable and device.history_enable and point.history_enable:
+        data = ModelUtils.row2dict_default(PointStoreModel.find_by_point_uuid(point.uuid))
+        point_store_history = PointStoreHistoryModel(**data)
+        point_store_history.save_to_db()
```

Comment:
I'm being picky now haha but this could be moved to a static method in `History service` or to `model_point_store.update` so that other drivers can use it as well (i.e. `modbus TCP`)

Reply:
Cool mate, this function should be on HistoryService instead here.